src/wallet/rpcwallet.cpp

```diff
@@ -4249,6 +4245,23 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         contextualTx.nVersion = 2; // Tx format should support vjoinsplits 
     }
 
+    // push sender address to the recipients to return change back to  the sender 
+    if(returnChangeToSenderAddr) {
+        int nMinDepth = 1;
+        if (fromTaddr) {
+            CAmount nBalance = getBalanceTaddr(fromaddress, nMinDepth, false);

```

Comment:
In the case we return change to sender address, a new operation will still be added to the global queue? Is that mean another transaction will be created?